src/tests/functional/inference_engine/transformations/common_optimizations/transpose_sinking_unary.cpp

```diff
@@ -0,0 +1,222 @@
+    CreateUnaryFactory<ov::opset9::Acos>(),
+    CreateUnaryFactory<ov::opset9::Asin>(),
+    CreateUnaryFactory<ov::opset9::Asinh>(),
+    CreateUnaryFactory<ov::opset9::Atan>(),
+    CreateUnaryFactory<ov::opset9::Ceiling>(),
+    CreateUnaryFactory<ov::opset9::Cos>(),
+    CreateUnaryFactory<ov::opset9::Cosh>(),
+    CreateUnaryFactory<ov::opset9::Erf>(),
+    CreateUnaryFactory<ov::opset9::Exp>(),
+    CreateUnaryFactory<ov::opset9::Gelu>(),
+    CreateUnaryFactory<ov::opset9::HSigmoid>(),
+    CreateUnaryFactory<ov::opset9::HSwish>(),
+    CreateUnaryFactory<ov::opset9::Log>(),
+    CreateUnaryFactory<ov::opset9::Negative>(),
+    CreateUnaryFactory<ov::opset9::Relu>(),
+    CreateUnaryFactory<ov::opset9::Sigmoid>(),
+    CreateUnaryFactory<ov::opset9::Sign>(),
+    CreateUnaryFactory<ov::opset9::Sin>(),
+    CreateUnaryFactory<ov::opset9::Sinh>(),
+    CreateUnaryFactory<ov::opset9::SoftSign>(),
+    CreateUnaryFactory<ov::opset9::Sqrt>(),
+    CreateUnaryFactory<ov::opset9::Tan>(),
+    CreateUnaryFactory<ov::opset9::Tanh>()
+};
+
+std::vector<size_t> unary_operations_numbers = {1, 10};
+
+} // namespace
+
+TEST_P(TransposeSinkingUnaryTestFixture, CompareFunctions) {
+    UnaryFactoryPtr unary_factory;
+    PassFactoryPtr pass_factory;
+    size_t num_unary_ops;
+    CreateGraphF model_factory;
+    CreateGraphF reference_model_factory;
+    std::tie(unary_factory, pass_factory, num_unary_ops, model_factory, reference_model_factory) = this->GetParam();
+
+    model = model_factory(unary_factory, num_unary_ops);
+    model_ref = reference_model_factory(unary_factory, num_unary_ops);
+    pass_factory->registerPass(manager);
+}
+
+INSTANTIATE_TEST_SUITE_P(TransposeSinkingUnaryForwardTestSuite, TransposeSinkingUnaryTestFixture,
+                         ::testing::Combine(::testing::ValuesIn(unary_factories),
+                                            ::testing::Values(CreatePassFactory<ov::pass::TransposeSinkingUnaryForward>()),
+                                            ::testing::ValuesIn(unary_operations_numbers),
+                                            ::testing::Values(CreateFunctionTransposeBefore),
+                                            ::testing::Values(CreateFunctionTransposeAfter)));
+
+INSTANTIATE_TEST_SUITE_P(TransposeSinkingUnaryBackwardTestSuite, TransposeSinkingUnaryTestFixture,
+                         ::testing::Combine(::testing::ValuesIn(unary_factories),
+                                            ::testing::Values(CreatePassFactory<ov::pass::TransposeSinkingUnaryBackward>()),
+                                            ::testing::ValuesIn(unary_operations_numbers),
+                                            ::testing::Values(CreateFunctionTransposeAfter),
+                                            ::testing::Values(CreateFunctionTransposeBefore)));
+
+INSTANTIATE_TEST_SUITE_P(TransposeSinkingUnaryBackward2ConsumersTestSuite, TransposeSinkingUnaryTestFixture,
```

Comment:
What `2` means in name?

Reply:
it means that transpose has 2 consumers. this test checks pattern restriction
Fixed it